code/modules/tgui/tgui_input/alert_input.dm

```diff
@@ -0,0 +1,146 @@
+/**
+ * Creates a TGUI alert window and returns the user's response.
+ *
+ * This proc should be used to create alerts that the caller will wait for a response from.
+ * Arguments:
+ * * user - The user to show the alert to.
+ * * message - The content of the alert, shown in the body of the TGUI window.
+ * * title - The of the alert modal, shown on the top of the TGUI window.
+ * * buttons - The options that can be chosen by the user, each string is assigned a button on the UI.
+ * * timeout - The timeout of the alert, after which the modal will close and qdel itself. Set to zero for no timeout.
+ * * autofocus - The bool that controls if this alert should grab window focus.
+ */
+/proc/tgui_alert(mob/user, message = "", title = "Alert", list/buttons = list("Ok"), timeout = 0, autofocus = TRUE, ui_state = GLOB.always_state)
+	if(!user)
+		user = usr
+
+	if(!istype(user))
+		if(!isclient(user))
+			return
+		var/client/client = user
+		user = client.mob
+
+	if(isnull(user.client))
+		return
+
+	// A gentle nudge - you should not be using TGUI alert for anything other than a simple message.
+	if(length(buttons) > 3)
+		log_tgui(user, "Error: TGUI Alert initiated with too many buttons. Use a list.", "TguiAlert")
+		return tgui_input_list(user, message, title, buttons, timeout)
+
+	// Client does NOT have tgui_input on: Returns regular input
+	if(user.client?.prefs?.toggles2 & PREFTOGGLE_2_DISABLE_TGUI_INPUT)
+		if(length(buttons) == 2)
+			return alert(user, message, title, buttons[1], buttons[2])
+		if(length(buttons) == 3)
+			return alert(user, message, title, buttons[1], buttons[2], buttons[3])
+
+	var/datum/tgui_alert/alert = new(user, message, title, buttons, timeout, autofocus, ui_state)
+
+	alert.ui_interact(user)
+	alert.wait()
+	if(alert)
+		. = alert.choice
+		qdel(alert)
+
+/**
+ * # tgui_alert
+ *
+ * Datum used for instantiating and using a TGUI-controlled modal that prompts the user with
+ * a message and has buttons for responses.
+ */
+/datum/tgui_alert
+	/// The title of the TGUI window
+	var/title
+	/// The textual body of the TGUI window
+	var/message
+	/// The list of buttons (responses) provided on the TGUI window
+	var/list/buttons
+	/// The button that the user has pressed, null if no selection has been made
+	var/choice
+	/// The time at which the tgui_alert was created, for displaying timeout progress.
+	var/start_time
+	/// The lifespan of the tgui_alert, after which the window will close and delete itself.
+	var/timeout
+	/// The bool that controls if this modal should grab window focus
+	var/autofocus
+	/// Boolean field describing if the tgui_alert was closed by the user.
+	var/closed
+	/// The TGUI UI state that will be returned in ui_state(). Default: always_state
+	var/datum/ui_state/state
+
+/datum/tgui_alert/New(mob/user, message, title, list/buttons, timeout, autofocus, ui_state)
```

Comment:
Why do we pass user here if we don't end up ever using that user arg?
Personal thing as well, but I prefer the `var = _var` syntax vs the `src.var = var`

Reply:
The argument is used to pass to the user who needs to open the interface